test/index.test.js

```diff
@@ -0,0 +1,455 @@
+    };
+
+    const buildIdConfig = {
+        buildId: '4b8d9b530d2e5e297b4f470d5b0a6e1310d29c5e'
+    };
+
+    const fakeCrumb = {
+        statusCode: 200,
+        body: {
+            _class: 'hudson.security.csrf.DefaultCrumbIssuer',
+            crumb: '24e80888069a1beaa5af3e0e3ef201d0',
+            crumbRequestField: 'Jenkins-Crumb'
+        }
+    };
+
+    const fakeJobInfo = {
+        lastBuild: {
+            number: 1
+        }
+    };
+
+    before(() => {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+    });
+
+    beforeEach(() => {
+        fsMock = {
+            readFileSync: sinon.stub()
+        };
+
+        jenkinsMock = sinon.stub();
+
+        requestMock = sinon.stub();
+
+        jenkinsClientMock = {
+            job: {
+                create: sinon.stub(),
+                get: sinon.stub(),
+                build: sinon.stub()
+            },
+            build: {
+                stop: sinon.stub(),
+                log: sinon.stub()
+            }
+        };
+
+        fsMock.readFileSync.returns(TEST_XML);
+
+        mockery.registerMock('fs', fsMock);
+        mockery.registerMock('jenkins', jenkinsMock);
+        mockery.registerMock('request', requestMock);
+
+        /* eslint-disable global-require */
+        Executor = require('../index');
+
+        executor = new Executor({
+            host: 'jenkins',
+            username: 'admin',
+            password: 'fakepassword'
+        });
+    });
+
+    afterEach(() => {
+        mockery.deregisterAll();
+        mockery.resetCache();
+    });
+
+    after(() => {
+        mockery.disable();
+    });
+
+    it('extends base class', () => {
+        assert.isFunction(executor.start);
+        assert.isFunction(executor.stream);
+        assert.isFunction(executor.stop);
+    });
+
+    describe('initJenkinsClient', () => {
+        it('return jenkinsClient object when request successes', (done) => {
+            const opts = {
+                baseUrl: executor.baseUrl,
+                headers: {
+                    [fakeCrumb.body.crumbRequestField]: fakeCrumb.body.crumb
+                }
+            };
+
+            jenkinsMock.returns(jenkinsClientMock);
+
+            executor.initJenkinsClient(JSON.stringify(fakeCrumb.body), (err, response) => {
+                assert.isNull(err);
+                assert.calledOnce(jenkinsMock);
+                assert.calledWith(jenkinsMock, opts);
+                assert.deepEqual(response, jenkinsClientMock);
+                done();
+            });
+        });
+
+        it('return error when jenkins() fails to instantiate a jenkins client', (done) => {
+            const error = new Error('Failed to instantiate jenkins client');
+
+            jenkinsMock.returns(null);
+
+            executor.initJenkinsClient(JSON.stringify(fakeCrumb.body), (err) => {
+                assert.deepEqual(err.message, error.message);
```

Comment:
If you're just checking strings, you can skip the Error object and just use the string directly.